src/common/ServerAPI.jsx

```diff
@@ -32,53 +32,42 @@ async function loginAccount(email, password) {
   const masterKey = await deriveMasterKey(email, password);
   const passwordHash = await deriveMasterPasswordHash(password, masterKey);
 
-  try {
-      const response = await axios.post(
-          LOGIN_ACCOUNT_URL,
-          {
-              email: email,
-              password: passwordHash,
-          },
-      );
-
-      const encryptedSymmetricKey = response.data.encrypted_symmetric_key;
-      const symmetricKey = await decryptSymmetricKey(encryptedSymmetricKey, masterKey);
-
-      return Promise.resolve(symmetricKey);
+  const response = await axios.post(
+    LOGIN_ACCOUNT_URL,
+    {
+        email: email,
+        password: passwordHash,
+    },
+  );
 
-  } catch(error) {
-      return Promise.reject(error);
-  }
+  const encryptedSymmetricKey = response.data.encrypted_symmetric_key;
+  return await decryptSymmetricKey(encryptedSymmetricKey, masterKey);
```

Comment:
Does the await happen before the return? I'm guessing this will this return the resolved value and not a promise in the pending state?

Reply:
Yes, it returns a Promise anyway since it's an async function but it will have the resolved value.